tests/unit/features/partitions/test_parts.py

```diff
@@ -130,39 +130,94 @@ def valid_filesets(self):
         return [
             "test",
             "(default)",
+            "(default)/",
+            "(default)//",
             "(default)/test",
+            "(default)//test",
             "(default)/test/(otherdir)",
             "(default)/test/(otherdir-123)",
             "(default)/test/(default)",
             "(default)/test/(kernel)",
+            "(default)/test/(a/b)",
+            "(default)/test/(a/c-d)",
             "test(default)",
             "test(default)/test",
             "test/(default)/test",
             "(kernel)",
+            "(kernel)/",
+            "(kernel)//",
             "(kernel)/test",
+            "(kernel)//test",
             "(kernel)/test/(otherdir)",
             "(kernel)/test/(otherdir-123)",
             "(kernel)/test/(default)",
             "(kernel)/test/(kernel)",
+            "(kernel)/test/(a/b)",
+            "(kernel)/test/(a/c-d)",
             "test/(kernel)",
             "test(kernel)/test",
+            "test/(kernel)/test",
+            "(a/b)",
+            "(a/b)/",
+            "(a/b)//",
+            "(a/b)/test",
+            "(a/b)//test",
+            "(a/b)/test/(otherdir)",
+            "(a/b)/test/(otherdir-123)",
+            "(a/b)/test/(default)",
+            "(a/b)/test/(kernel)",
+            "(a/b)/test/(a/b)",
+            "(a/b)/test/(a/c-d)",
+            "test/(a/b)",
+            "test(a/b)/test",
+            "test/(a/b)/test",
+            "(a/c-d)",
+            "(a/c-d)/",
+            "(a/c-d)//",
+            "(a/c-d)/test",
+            "(a/c-d)//test",
+            "(a/c-d)/test/(otherdir)",
+            "(a/c-d)/test/(otherdir-123)",
+            "(a/c-d)/test/(default)",
+            "(a/c-d)/test/(kernel)",
+            "(a/c-d)/test/(a/b)",
+            "(a/c-d)/test/(a/c-d)",
+            "test/(a/c-d)",
+            "test(a/c-d)/test",
+            "test/(a/c-d)/test",
```

Comment:
Interesting list of cases, why is `(a/c-d)/...` considered valid? Is it because it considers `(a/c-d)` as a directory name? Or are `(a` and `c-d)` treated as separate directories?

Reply:
The latter is correct.  Does [Alex's comment](https://github.com/canonical/craft-parts/pull/600/files/6ea1343da63b06c762256fdda9a925986d103b82#r1419456295) answer your question?

Reply:
Yes, thanks for the clarification.